### ISSUE TITLE
Unsatisfied dependency error

### ISSUE THREAD

**Issue by bachii1010** [Microsoft/AL#1786]
Hi,

we have added test.app as dependency and we are getting the below error while publishing the app:
![image](https://user-images.githubusercontent.com/26792519/37399252-2fc93960-27a7-11e8-95e7-9d37d669d45a.png)

Please suggest us.

Thanks & regards
Varun Reddy

**Comment by atoader** [comment]
What version of the symbols do you have published on the server?

**Comment by bachii1010** [comment]
Hi @atoader 

What do you mean by **server**
which server it is u mean?

**Comment by atoader** [comment]
Sorry for the confusion. Let me rephrase: What version of NAV are you using? What version of the AL Language extension are you using? Can you reproduce the problem targetting different servers? Do you have a minimal code sample that reproduces this?

**Comment by bachii1010** [comment]
Hi,

We are using NAV 2018 CU2 and the AL extension is 0.12.16641.

i don't have a minimal solution because the error is coming while publishing the entire app .

Please suggest.

**Comment by atoader** [comment]
Hi! I see that you are using a version of NAV that is in mainstream support. This GitHub repository is dedicated to handling issues and feedback for the latest Developer Preview releases of the AL development tools for Dynamics 365 "Tenerife". Please open a Support Request to CSS through the PartnerSource portal or contact your Service Account Manager (SAM) in the local subsidiary. The CSS team is well suited for helping you reproduce this issue and backporting the eventual fix to your version of NAV.

**Comment by bachii1010** [comment]
@atoader  

I have checked the same thing in march devpreview update . i got the same error i have described while publishing.

please suggest

**Comment by StanislawStempin** [comment]
Can you please run the Get-NavAppInfo -SymbolsOnly commandlet on your server and see if Test and System packages with the versions equal or higher to the ones specified in your app,json are present?

**Comment by bachii1010** [comment]
@StanislawStempin  see below both are same:

![image](https://user-images.githubusercontent.com/26792519/38552544-08dcb532-3cda-11e8-84cb-2c4dfe4b2eb0.png)

Please suggest

**Comment by StanislawStempin** [comment]
Your app.json seems to be the problem.  Test and System packages should not be specified in the `dependencies` element. Instead they are covered by app.json elements:
```
"platform" : "12.0.0.0"
"test" : "12.0.0.0"

Please try to change the app.json and try again. Have you got this by using the Designer to generate the contents of app.json?
```

**Comment by atoader** [comment]
Hi @bachii1010 ! Could you let us know if the previous suggestion worked?

**Comment by bachii1010** [comment]
@StanislawStempin @atoader  yes the above suggestion is working for me thanks.
